fix: relocate Automations snippet includes and fix localized imports (#2727)
wandb/docs: models/sweeps/define-sweep-configuration.mdx

</Tabs>

> The top-level `parameters` key nests the following keys: `learning_rate`, `batch_size`, `epoch`, and `optimizer`. For each nested key you specify, you can provide one or more values, a distribution, a probability, and more. For more information, see the [parameters](./sweep-config-keys#parameters) section in [Sweep configuration options](./sweep-config-keys).

Reviewer: More off syntax highlighting in the diff view might point to something wrong in the formatting.

Author: Stripped trailing whitespace and normalized spacing. The residual "off" diff highlighting is from Mintlify's `title=` fence attribute (valid syntax; renders fine).

## Double nested parameters

Use nested parameters when you want to group related hyperparameters together or when your training code expects a nested configuration structure. To define a nested parameter, include an additional `parameters` key under the top-level parameter name.

> The following example shows a sweep configuration with three nested parameters: `nested_category_1`, `nested_category_2`, and `nested_category_3`. Each nested parameter includes two additional parameters: `momentum` and `weight_decay`.

Reviewer: "three nested parameters" is redundant with the actual parameter names and represents a tiny amount of tech debt if someone adds a parameter but doesn't remember to update the count.

Author: Dropped the redundant "three"/"two" counts so the prose can't drift from the example.

<Note>

> `nested_category_1`, `nested_category_2`, and `nested_category_3` are placeholders. Replace them with names that fit your use case.

Reviewer: Our users are smarter than this note, I think.

Author: Removed the placeholder Note.

    },
}

sweep_id = wandb.sweep(sweep=sweep_configuration, project="[PROJECT]")
wandb.agent(sweep_id, function=main, count=4)
```

During a sweep run, `run.config["nested_param"]` reflects the subtree defined by the

> sweep (`learning_rate`, `double_nested_param`) config and doesn't include `manual_key` defined

Reviewer: This reads weird with "sweep config" split up across the parenthetical, and "doesn't include .... defined" maybe should be "doesn't define"?

Author: Reworded so "sweep configuration" isn't split, and "…doesn't include `manual_key`, which is defined in `wandb.init(config=...)`."

in `wandb.init(config=...)`.
</Warning>

## Sweep configuration template

Use this template as a starting point when authoring a new sweep configuration. It illustrates the most common parameter and early-termination patterns so you can copy it and fill in the values for your own search.

The following template shows how you can configure parameters and specify search constraints. Replace `hyperparameter_name` with the name of your hyperparameter and any values enclosed in brackets.

```yaml title="config.yaml"
program: [INSERT]
method: [INSERT]

> parameter:

Author: Fixed the template top-level key `parameter:` → `parameters:`.

- ${Command macro}      
```

To express a numeric value using scientific notation, add the YAML `!!float` operator, which casts the value to a floating-point number. For example, `min: !!float 1e-5`. For more information, see [Macro and custom command arguments example](#macro-and-custom-command-arguments-example).

## Sweep configuration examples

> The following examples show complete sweep configurations for common scenarios. Use them as references when adapting a sweep to your own training script.

Reviewer: ```suggestion
The following sweep configurations illustrate common scenarios. Use them as references when adapting a sweep to your own training script.
```

Author: Applied.

### Bayes hyperband example

> The following example combines Bayesian search with Hyperband early termination so that underperforming runs stop early, freeing resources for more promising configurations.

Reviewer: ```suggestion
The following example combines Bayesian search with Hyperband early termination to stop underperforming runs stop and preserve resources for more promising configurations.
```

Author: Applied with a tiny grammar fix (the suggestion had a doubled "stop"): "…to stop underperforming runs early and preserve resources…."

  - "--optimizer=${optimizer}"
  - "--test=True"
```

> The associated Python script (`run.py`) can then parse these command-line arguments using the `argparse` module:

Reviewer: I think a newline after line 458 would fix the diff syntax highlighting in 459.

Author: Added the blank line after the fenced block.
